### PR TITLE
Add GitHub action for CI tests

### PR DESCRIPTION
**Pull request by ssiccha**
This PR configures GitHub Actions to run the package tests. It was created by a script. You can view the results of the added GitHub action at:
https://github.com/ssiccha/io/actions
This PR contains four commits which do:
- adds a file `.github/workflows/CI.yml` which configures GitHub Actions to run the package tests,
- removes the `.travis.yml` file if it exists in the package,
- adds a CI action badge to the README.md, if it found none, and
- removes the travis CI badge from the README.md, if it finds one.

Since this PR is generated automatically, please double-check that the last three commits do not break anything. You may also have to specify values for the variables `GAP_PKGS_TO_BUILD` and `GAP_PKGS_TO_CLONE` in the `.github/workflows/CI.yml` file as described further below.

The added GitHub action `CI` contains a job `test`, which runs your package's tests using the same scripts as previously did Travis, namely those from
https://github.com/gap-system/pkg-ci-scripts
By default it tests your package with the following gap branches:
- master
- stable-4.11
- stable-4.10

The action `CI` also contains a job `manual`, which compiles your package's documentation with latex and uploads the resulting pdf as an artifact, which means that it can be downloaded once all jobs of the action completed.

If you need to compile packages to load your package, set the input `GAP_PKGS_TO_BUILD` for the action `gap-actions/setup-gap-for-packages@v1` to a space-separated list of packages as follows:
```
- uses: gap-actions/setup-gap-for-packages@v1
  with:
    GAPBRANCH: ${{ matrix.gap-branch }}
    GAP_PKGS_TO_BUILD: "<list-of-packages>"
```

If you need to clone the development version of packages set the input `GAP_PKGS_TO_CLONE` for the action `gap-actions/setup-gap-for-packages@v1` to a space-separated list of packages as follows:
```
- uses: gap-actions/setup-gap-for-packages@v1
  with:
    GAPBRANCH: ${{ matrix.gap-branch }}
    GAP_PKGS_TO_CLONE: "<list-of-packages>"
```

Notice that you may also have to set inputs for the action `gap-actions/setup-gap-for-packages@v1` in the job `manual`.

If you want to adjust which tests are run have a look at the [documentation of the job matrix](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix).You can add changes to this PR by adding my fork as a remote, checking it out and then committing and pushing as follows:
```
git remote add ssiccha https://github.com/ssiccha/io
git checkout ss/add-GitHub-actions-for-package-tests
# Add and commit your changes
...
git push
```

The custom gap actions which are used by the added action can be found at:
https://github.com/gap-actions

### CODE REVIEW

**Comment by ssiccha** [comment]
I'm still trying to get the 32bit tests to pass.

**Comment by ssiccha** [comment]
@fingolfin The `ABI=32` run is failing: https://github.com/ssiccha/io/runs/1412602919?check_suite_focus=true#step:3:1013

I've compared the run with a successful one from travis. The line that fails in the action looks like this on travis:
```
/bin/bash /home/travis/gap/libtool --mode=compile gcc -m32 -g -O2 -MQ gen/src/io.lo -MMD -MP -MF gen/src/io.d -Igen --coverage -m32 -o gen/src/io.lo -I/home/travis/gap/build -I/home/travis/gap/src -I/home/travis/gap -DHAVE_CONFIG_H -c src/io.c
```
https://travis-ci.com/github/gap-packages/io/jobs/392772295#L1167
and like this in the action run:
```
/bin/bash /home/runner/gap/libtool --mode=compile gcc -g -O2 -MQ gen/src/io.lo -MMD -MP -MF gen/src/io.d -Igen --coverage -m32 -o gen/src/io.lo -I/home/runner/gap/build -I/home/runner/gap/src -I/home/runner/gap -DHAVE_CONFIG_H -c src/io.c
```
https://github.com/ssiccha/io/runs/1412602919?check_suite_focus=true#step:3:1013
So it seems there is an `-m32` missing directly after the `compile gcc` bit. Do you have any idea what the reason could be for this?

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            ABI: 32
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: gap-actions/setup-gap-for-packages@v1
+        with:
+          GAPBRANCH: ${{ matrix.gap-branch }}
+          GAP_PKGS_TO_BUILD: ${{ matrix.GAP_PKGS_TO_BUILD }}
+          HPCGAP: ${{ matrix.HPCGAP }}
+          ABI: ${{ matrix.ABI }}
```
Comment (fingolfin) [inline .github/workflows/CI.yml:41]: Alphabetic order makes longterm maintenance easier (but more importantly, it should allow me to re-trigger a build of this PR ;-) )
```suggestion
          ABI: ${{ matrix.ABI }}
          GAPBRANCH: ${{ matrix.gap-branch }}
          GAP_PKGS_TO_BUILD: ${{ matrix.GAP_PKGS_TO_BUILD }}
          HPCGAP: ${{ matrix.HPCGAP }}
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  manual:
+    name: Build manuals
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: gap-actions/setup-gap-for-packages@v1
+      - uses: gap-actions/compile-documentation-for-packages@v1
+        with:
+          use-latex: 'true'
```
Comment (fingolfin) [inline .github/workflows/CI.yml:57]:
```suggestion
          use-latex: true
```

**Comment by ssiccha** [comment]
32 bit tests work now.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-[![Build Status](https://travis-ci.com/gap-packages/io.svg?branch=master)](https://travis-ci.com/gap-packages/io)
+![CI](https://github.com/ssiccha/io/workflows/CI/badge.svg)
```
Comment (fingolfin) [inline README.md:1]: That needs to be adjusted. Also there shouldn't be just a badge, it should also link to the CI results for this workflow on the master branch (as I did for e.g. gap-system/gap).

**Review by fingolfin — CHANGES_REQUESTED** [review]
Also perhaps squash this into a single commit?

**Comment by fingolfin** [comment]
@ssiccha do you plan to finish this or should I take over? I also don't quite understand how this relates to PR #89 resp. vice versa...

**Comment by alex-konovalov** [comment]
What happened to this and similar PRs elsewhere? Is there a canonical setup others may use? It would be useful to have a canonical setup in the example package https://github.com/gap-packages/example/pulls ...

File: README.md (hunk 3 above)
Comment (fingolfin) [reply README.md:1]: Ping? @ssiccha any plans to update this PR?

Comment (ssiccha) [reply README.md]: PR is updated now. The badge now is:
```
[![CI](https://github.com/ssiccha/io/workflows/CI/badge.svg)](https://github.com/gap-packages/recog/actions?query=workflow%3ACI+branch%3Amaster)
```

**Comment by ssiccha** [comment]
@alex-konovalov Re the other PRs: I didn't have the time to finish them. :/
Re the example package: I think that's a good idea. @fingolfin Would it make sense to extend `PackageMaker` so that `PackageWizard` also offers to create a GitHub action for CI testing?

**Comment by ssiccha** [comment]
Tests work. See https://github.com/ssiccha/io/actions

**Comment by ssiccha** [comment]
Updated the Codecov badge and squashed.

**Comment by ssiccha** [comment]
@fingolfin Could you have a look again?

**Pull request closed without merging**